server/middleware/checkAuth.js

```
@@ -0,0 +1,21 @@
+const jwt = require('jsonwebtoken');
+
+const checkAuth = async (req, res, next) => {
+  try {
+    // Get the token from the header
+    const token = req.header('Authorization')?.replace('Bearer ', '');
+
+    if (!token) {
+      throw new Error('No token found');
+    }
+
+    // Verify the token
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    req.userId = decoded.userId;
+    next();
+  } catch (error) {
+    res.status(401).json({error: 'Auth failed'})
```

The error can up from somewhere else in the call stack e.g. the call to req.header() might yield its own exception, so we should return the error message body or context from the error itself.